src/components/Button/Button.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+
+import _s from './Button.css';
+
+class Button extends React.Component {
+  _mouseEnter(...a) {
+    if(this.props.onMouseEnter)
+      this.props.onMouseEnter(...a);
+  }
+
+  _mouseLeave(...a) {
+    if(this.props.onMouseLeave)
+      this.props.onMouseLeave(...a);
+  }
+
+  render() {
+    const { children, text, ...rest } = this.props;
+    
+    return (
+      <button 
+        {...rest}
+        className={_s.button} 
+        onMouseEnter={(...a) => this._mouseEnter(...a)}
+        onMouseLeave={(...a) => this._mouseLeave(...a)}
```

Comment:
IMO `onMouseEnter` and `onMouseLeave` doesn't belong on Button. 
Also I think this is pretty much the same as doing `onMouseEnter={this.props.mouseEnter}`?

Either way IconButton can add hooks for this be simply passing `onMouseEnter` as a prop using `...rest`.